lib/src/value/calculation.dart

```diff
@@ -158,6 +382,200 @@ class SassCalculation extends Value {
     return SassCalculation._("clamp", args);
   }
 
+  /// Creates a `pow()` calculation with the given [base] and [exponent].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value pow(Object base, Object? exponent) {
+    var args =
+        List<Object>.unmodifiable([base, if (exponent != null) exponent]);
+    _verifyLength(args, 2);
+    base = _simplify(base);
+    exponent = exponent.andThen(_simplify);
+    if (base is! SassNumber ||
+        exponent is! SassNumber ||
+        exponent.hasUnit('%') ||
+        base.hasUnit('%')) {
+      return SassCalculation._("pow", args);
+    }
+    _verifyCompatibleNumbers([base, exponent]);
+    return number_lib.pow(base, exponent);
+  }
+
+  /// Creates a `log()` calculation with the given [number] and [base].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// If arguments contains exactly a single argument,
+  /// the base is set to `math.e` by default.
+  static Value log(Object number, Object? base) {
+    number = _simplify(number);
+    base = base.andThen(_simplify);
+    var args = [number, if (base != null) base];
+    if (number is! SassNumber ||
+        number.hasUnit('%') ||
+        (base is SassNumber && base.hasUnit('%'))) {
+      return SassCalculation._("log", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return number_lib.log(number, base is SassNumber ? base : null);
```

Comment:
What if `base` is a string or an interpolation? You don't want to treat that as null.

This is something that should have a test.

Reply:
Hm those cases were in the tests and working 🤔. Still, I added another edge case to the if statement. Thanks!

Reply:
Oh yes, later on I found out it was passing because the first argument had a '%'. So it covered the test case to return a calculation. If I did `log(2, test);` it would result in `b: 0.6931471806;`. So yes, I was missing a test for that edge case and the verification of `(base is SassNumber && base.hasUnit('%')` was missing.